packages/evm/test/Enclave.spec.ts

```diff
@@ -36,19 +37,61 @@ const proof = "0x1337";
 // Hash function used to compute the tree nodes.
 const hash = (a: bigint, b: bigint) => poseidon2([a, b]);
 
+type SetupConfig = {
+  inputLimit: number;
+};
+
+async function extractEventLog(name: string, tx: ContractTransactionResponse) {
+  const receipt = await tx.wait();
+
+  if (receipt == null) throw new Error("Receipt not returned");
+  const event = receipt.logs.find(
+    (log) => log instanceof EventLog && log.fragment.name === name,
+  ) as EventLog | undefined;
+  if (!event) throw new Error("Event not found");
+  return event;
+}
+
+function defaultSetupConfig(config: Partial<SetupConfig> = {}): SetupConfig {
+  return {
+    inputLimit: 0,
+    ...config,
+  };
+}
+
+async function deployInputValidatorContracts() {
+  const inputValidatorChecker = await deployInputValidatorCheckerFixture();
+  const inputValidatorPolicyFactory =
+    await deployInputValidatorPolicyFactoryFixture();
+
+  return {
+    inputValidatorChecker,
+    inputValidatorPolicyFactory,
+  };
+}
+
 describe("Enclave", function () {
-  async function setup() {
+  async function setup(config = defaultSetupConfig()) {
     const [owner, notTheOwner] = await ethers.getSigners();
 
     const poseidon = await PoseidonT3Fixture();
     const registry = await deployCiphernodeRegistryFixture();
     const decryptionVerifier = await deployDecryptionVerifierFixture();
     const computeProvider = await deployComputeProviderFixture();
-    const inputValidator = await deployInputValidatorFixture();
+
+    const { inputValidatorChecker, inputValidatorPolicyFactory } =
+      await deployInputValidatorContracts();
+
     const e3Program = await deployE3ProgramFixture(
-      await inputValidator.getAddress(),
+      await inputValidatorPolicyFactory.getAddress(),
+      await inputValidatorChecker.getAddress(),
     );
 
+    // TODO: is this too restrictive?
```

Comment:
note: can you expand this todo?

Reply:
I added some more context around what I was thinking.